rust/src/dataset.rs

```diff
@@ -248,7 +248,12 @@ impl Dataset {
     /// Check out the specified version of this dataset
     pub async fn checkout_version(&self, version: u64) -> Result<Self> {
         let base_path = self.base.clone();
-        let manifest_file = manifest_path(&base_path, version);
```

Comment:
Should the `manifest_path` and `latest_manifest_path` functions become private functions?  It seems these are no longer safe to call (and you should always go through the commit handler)

Reply:
done as part of #1198